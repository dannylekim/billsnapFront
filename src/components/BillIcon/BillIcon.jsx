import {FaBus, FaCar, FaQuestion, FaShoppingBag, FaShoppingCart, FaUtensils,} from "react-icons/fa/index";
import React from "react";

const BillIcon = ({category}) => {
  const color = "rgba(0, 0, 0, 0.96)";
  switch (category.toLowerCase()) {
    case "food":
      return <FaUtensils color={color} size={24} />;
    case "transport":
      return <FaCar color={color} size={24} />;
    case "public-transport":
      return <FaBus color={color} size={24} />;
    case "grocery":
      return <FaShoppingCart color={color} size={24} />;
    case "shopping":
      return <FaShoppingBag color={color} size={24} />;
    default:
      return <FaQuestion color={color} size={24} />;
  }
};

export default BillIcon;
