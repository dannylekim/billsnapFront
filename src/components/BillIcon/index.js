import BillIcon from "./BillIcon";

export default BillIcon;