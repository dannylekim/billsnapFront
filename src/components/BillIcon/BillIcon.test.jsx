import BillIcon from "./BillIcon";
import React from "react";

describe("BillIcon", () => {
  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("BillIcon should match food snap shot", () => {
        matches(<BillIcon category="food" />);
      });
      it("BillIcon should match transport snap shot", () => {
        matches(<BillIcon category="transport" />);
      });
      it("BillIcon should match public-transport snap shot", () => {
        matches(<BillIcon category="public-transport" />);
      });
      it("BillIcon should match grocery snap shot", () => {
        matches(<BillIcon category="grocery" />);
      });
      it("BillIcon should match shopping snap shot", () => {
        matches(<BillIcon category="shopping" />);
      });
      it("BillIcon should match ? snap shot", () => {
        matches(<BillIcon category="?" />);
      });
    });
  });
});
