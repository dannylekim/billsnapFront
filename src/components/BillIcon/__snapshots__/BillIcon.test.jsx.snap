// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillIcon render snapshots ðŸ“¸ BillIcon should match ? snap shot 1`] = `
<svg
  color="rgba(0, 0, 0, 0.96)"
  fill="currentColor"
  height={24}
  size={24}
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.96)",
    }
  }
  viewBox="0 0 384 512"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
  />
</svg>
`;

exports[`BillIcon render snapshots ðŸ“¸ BillIcon should match food snap shot 1`] = `
<svg
  color="rgba(0, 0, 0, 0.96)"
  fill="currentColor"
  height={24}
  size={24}
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.96)",
    }
  }
  viewBox="0 0 416 512"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M207.9 15.2c.8 4.7 16.1 94.5 16.1 128.8 0 52.3-27.8 89.6-68.9 104.6L168 486.7c.7 13.7-10.2 25.3-24 25.3H80c-13.7 0-24.7-11.5-24-25.3l12.9-238.1C27.7 233.6 0 196.2 0 144 0 109.6 15.3 19.9 16.1 15.2 19.3-5.1 61.4-5.4 64 16.3v141.2c1.3 3.4 15.1 3.2 16 0 1.4-25.3 7.9-139.2 8-141.8 3.3-20.8 44.7-20.8 47.9 0 .2 2.7 6.6 116.5 8 141.8.9 3.2 14.8 3.4 16 0V16.3c2.6-21.6 44.8-21.4 48-1.1zm119.2 285.7l-15 185.1c-1.2 14 9.9 26 23.9 26h56c13.3 0 24-10.7 24-24V24c0-13.2-10.7-24-24-24-82.5 0-221.4 178.5-64.9 300.9z"
  />
</svg>
`;

exports[`BillIcon render snapshots ðŸ“¸ BillIcon should match grocery snap shot 1`] = `
<svg
  color="rgba(0, 0, 0, 0.96)"
  fill="currentColor"
  height={24}
  size={24}
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.96)",
    }
  }
  viewBox="0 0 576 512"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
  />
</svg>
`;

exports[`BillIcon render snapshots ðŸ“¸ BillIcon should match public-transport snap shot 1`] = `
<svg
  color="rgba(0, 0, 0, 0.96)"
  fill="currentColor"
  height={24}
  size={24}
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.96)",
    }
  }
  viewBox="0 0 512 512"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M488 128h-8V80c0-44.8-99.2-80-224-80S32 35.2 32 80v48h-8c-13.25 0-24 10.74-24 24v80c0 13.25 10.75 24 24 24h8v160c0 17.67 14.33 32 32 32v32c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32v-32h192v32c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32v-32h6.4c16 0 25.6-12.8 25.6-25.6V256h8c13.25 0 24-10.75 24-24v-80c0-13.26-10.75-24-24-24zM112 400c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32zm16-112c-17.67 0-32-14.33-32-32V128c0-17.67 14.33-32 32-32h256c17.67 0 32 14.33 32 32v128c0 17.67-14.33 32-32 32H128zm272 112c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
  />
</svg>
`;

exports[`BillIcon render snapshots ðŸ“¸ BillIcon should match shopping snap shot 1`] = `
<svg
  color="rgba(0, 0, 0, 0.96)"
  fill="currentColor"
  height={24}
  size={24}
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.96)",
    }
  }
  viewBox="0 0 448 512"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M352 160v-32C352 57.42 294.579 0 224 0 153.42 0 96 57.42 96 128v32H0v272c0 44.183 35.817 80 80 80h288c44.183 0 80-35.817 80-80V160h-96zm-192-32c0-35.29 28.71-64 64-64s64 28.71 64 64v32H160v-32zm160 120c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zm-192 0c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24z"
  />
</svg>
`;

exports[`BillIcon render snapshots ðŸ“¸ BillIcon should match transport snap shot 1`] = `
<svg
  color="rgba(0, 0, 0, 0.96)"
  fill="currentColor"
  height={24}
  size={24}
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.96)",
    }
  }
  viewBox="0 0 512 512"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M499.99 176h-59.87l-16.64-41.6C406.38 91.63 365.57 64 319.5 64h-127c-46.06 0-86.88 27.63-103.99 70.4L71.87 176H12.01C4.2 176-1.53 183.34.37 190.91l6 24C7.7 220.25 12.5 224 18.01 224h20.07C24.65 235.73 16 252.78 16 272v48c0 16.12 6.16 30.67 16 41.93V416c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32v-32h256v32c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32v-54.07c9.84-11.25 16-25.8 16-41.93v-48c0-19.22-8.65-36.27-22.07-48H494c5.51 0 10.31-3.75 11.64-9.09l6-24c1.89-7.57-3.84-14.91-11.65-14.91zm-352.06-17.83c7.29-18.22 24.94-30.17 44.57-30.17h127c19.63 0 37.28 11.95 44.57 30.17L384 208H128l19.93-49.83zM96 319.8c-19.2 0-32-12.76-32-31.9S76.8 256 96 256s48 28.71 48 47.85-28.8 15.95-48 15.95zm320 0c-19.2 0-48 3.19-48-15.95S396.8 256 416 256s32 12.76 32 31.9-12.8 31.9-32 31.9z"
  />
</svg>
`;
